### PR TITLE
Add pagination functionality

### PR DESCRIPTION
**Pull request by MagoDelBlocco**
This PR adds the structure needed for changing the pages shown on the front end. **This is done via 2 buttons "prevPage" and "nextPage", and a visual indicator of the current page number.**

Currently, each page has a fixed number of **20 build snapshots**, but the plan is for the user to be able to choose how many entries per page it wants. The styling will come in a later PR.

### CODE REVIEW

File: src/main/webapp/components/BuildSnapshotContainer.js [hunk 1]
```diff
@@ -43,10 +52,27 @@ export const BuildSnapshotContainer = React.memo((props) =>
           No new revisions to display as of {new Date().toLocaleString()}.</span>;
     }
 
+    let hasPrev = pageNumber == 0;
```
Comment (samuelsogbesan) [inline src/main/webapp/components/BuildSnapshotContainer.js:55]: Use <code>===</code> for [equality checks](https://google.github.io/styleguide/jsguide.html#features-equality-checks)
```suggestion
    let hasPrev = pageNumber === 0;
```

**Review by samuelsogbesan — APPROVED** [review]
LGTM, bar one nit.

I'd recommend we move the pagination-navbar to render conditionally with the build-snaphot-container. This way if no data is returned, we don't still have the pagination section. Also, it would be better if the pagination tooling came above the build-snapshot-container, otherwise it will get pushed to the bottom and won't be visible when we start getting data in.

This change can be for a future PR because it'll require the logic to shuffle around a little bit. 

i'd also recommend in the future to just add the paginationNavbar directly into the div rather than setting it up externally, for readability.

Good job though :)

**Comment by MagoDelBlocco** [comment]
> I'd recommend we move the pagination-navbar to render conditionally with the build-snaphot-container. This way if no data is returned, we don't still have the pagination section. 

Theoretically, this app should always have data once we properly deploy it, and the only cases where we don't have any data to show would be in test builds and now, in the beginning. I don't think this warrants the addition of new logic just for this case.

**Comment by MagoDelBlocco** [comment]
> I'd recommend we move the pagination-navbar to render conditionally with the build-snaphot-container. This way if no data is returned, we don't still have the pagination section. 

Addressed this by making the buttons invisible instead of grey-ed out, in #193. No additional logic needs to be added.

**Comment by MagoDelBlocco** [comment]
Opened issue #208, which aims to move the Pagination structure to its own component.

File: src/main/webapp/components/BuildSnapshotContainer.js [hunk 2]
```diff
@@ -9,18 +9,27 @@ import {BuildSnapshot} from "./BuildSnapshot";
...
 export const BuildSnapshotContainer = React.memo((props) =>
   {
-    // Number of entries to be pulled.
-    // TODO: Determine true value for number of entries.
-    const NUMBER = 2;
+    const [pageNumber, setPageNumber] = React.useState(0);
+    const [data, setData] = React.useState([]);
+
+    // Number of entries to be pulled. 20 seems reasonable enough.
+    // TODO: Add options for this number on paginationNavbar
```
Comment (Somae) [inline src/main/webapp/components/BuildSnapshotContainer.js:16]:
```suggestion
    // TODO: Add options for this number on paginationNavbar.
```

File: src/main/webapp/components/BuildSnapshotContainer.js [hunk 3]
```diff
@@ -9,18 +9,27 @@ import {BuildSnapshot} from "./BuildSnapshot";
...
-    // TODO: vary multiplier based on page number.
-    const OFFSET = NUMBER * 0;
+    const OFFSET = NUMBER * pageNumber;
 
+    // The endpoint from where we get data from
     const SOURCE = `/builders/number=${NUMBER}/offset=${OFFSET}`;
 
-    const [data, setData] = React.useState([]);
+    /**
+     * Moves pageNumber with a set increment
```
Comment (Somae) [inline src/main/webapp/components/BuildSnapshotContainer.js:27]:
```suggestion
     * Moves pageNumber with a set increment.
```

File: src/main/webapp/components/BuildSnapshotContainer.js [hunk 4]
```diff
@@ -9,18 +9,27 @@ import {BuildSnapshot} from "./BuildSnapshot";
...
+    // TODO: Add options for this number on paginationNavbar
+    const NUMBER = 20;
 
     // The starting point from which the entries are selected.
     // Relative to the starting index.
-    // TODO: vary multiplier based on page number.
-    const OFFSET = NUMBER * 0;
+    const OFFSET = NUMBER * pageNumber;
 
+    // The endpoint from where we get data from
```
Comment (Somae) [inline src/main/webapp/components/BuildSnapshotContainer.js:23]:
```suggestion
    // The endpoint from where we get data from.
```

**Review by Somae — APPROVED** [review]

**Comment by MagoDelBlocco** [comment]
This PR got "emergency merged" to provide the functionality needed for the live demo. It was added in the list at #214 for PRs that need readability re-review.

**Pull request merged**